docs: Add headers for easier Core, Base and stdlib reference

adigitoleo opened the pull request:
This adds one new section (h2) and a nested subsection (h3) to the landing page of the Julia documentation website and associated PDF file. The new section provides a quick overview of the differences between `Core`, `Base` and the standard library, as well as a list of links to standard library packages.

The list of standard library packages is thus duplicated in the website sidebar, but I don't think this is an issue because the sidebar is primarily intended for navigation, whereas this list is primarily intended to be used for referencing purposes.

Partially addresses #28712, although I don't provide short descriptions of each standard library package. I think the appropriate place for that would be in module-level docstrings of the packages themselves, which are not currently available.

adigitoleo committed:
docs: Add headers for easier Core, Base and stdlib reference

This adds one new section (h2) and a nested subsection (h3) to the
landing page of the Julia documentation website and associated PDF file.
The new section provides a quick overview of the differences between
`Core`, `Base` and the standard library, as well as a list of links to
standard library packages.

The list of standard library packages is thus duplicated in the website
sidebar, but I don't think this is an issue because the sidebar is
primarily intended for navigation, whereas this list is primarily
intended to be used for referencing purposes.

Partially addresses #28712, although I don't provide short descriptions
of each standard library package. I think the appropriate place for that
would be in module-level docstrings of the packages themselves, which
are not currently available.

adigitoleo commented:
There has been interest in the linked issue in backporting this (to 1.9 or 1.10). If accepted I can cherry-pick this onto one of those release branches and open separate PRs for those.

adigitoleo commented:
The linked issue also contains a screenshot of how this looks in the web page. See https://github.com/JuliaLang/julia/issues/28712#issuecomment-2258486991

waldyrious commented:
I've added a [comment](https://github.com/JuliaLang/julia/issues/28712#issuecomment-2261537263) to the associated issue with a suggestion for the section heading and some notes that may be helpful to expand the info added here. In any case, this is great work already, and definitely an improvement over the current situation!

adigitoleo committed:
docs: Better integration of new stdlib docs with existing Base docs

adigitoleo committed:
fix: Fix typo @id -> @ref

waldyrious commented on doc/src/index.md:
````diff
@@ -127,19 +127,25 @@ language. In addition to the above, some advantages of Julia over comparable sys
 
 ```@docs; canonical=false
 Core
 Base
````
I assume it's deliberate, but just to be sure: why don't you include `Main` here too?

adigitoleo replied:
My thinking was that `Main` is not really a namespace where people should be looking for either Julia builtin symbols or "standard library" packages/modules, because the symbols under `Main` are user-defined. So, to keep this section as a reference for "what do I get out of the box", I omitted it. I don't feel very strongly about it, but in some ways I didn't like the aesthetic of just duplicating the Standard Modules section from `base.md`.

waldyrious replied:
Fair enough!

adigitoleo committed:
docs: Add list of stdlib _jll packages

Addresses <https://github.com/JuliaLang/julia/issues/28712#issuecomment-2266204391>.

ViralBShah committed:
Merge branch 'master' into docs-stdlib-section
